Content.Server/_CorvaxGoob/Deathmatch-CS/CSRuleSystem.cs

```diff
@@ -0,0 +1,165 @@
+using Content.Server.Administration.Systems;
+using Content.Server.GameTicking;
+using Content.Server.GameTicking.Rules;
+using Content.Server.Ghost.Roles.Components;
+using Content.Server.Maps;
+using Content.Shared.GameTicking;
+using Content.Shared.GameTicking.Components;
+using Content.Shared.Mind.Components;
+using Content.Shared.Mobs;
+using Content.Shared.Mobs.Systems;
+using Robust.Server.GameObjects;
+using Robust.Shared.Map;
+using Robust.Shared.Player;
+using Robust.Shared.Random;
+using System.Linq;
+
+namespace Content.Server._CorvaxGoob.Deathmatch_CS;
+
+public sealed class Session
+{
+    public MapId MapId;
+    public List<EntityUid> Players = new();
+}
+
+public sealed class CSRuleSystem : GameRuleSystem<CSRuleComponent>
+{
+    [Dependency] private readonly MapSystem _map = default!;
+    [Dependency] private readonly IGameMapManager _gameMapManager = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly MobStateSystem _mobStateSystem = default!;
+
+    private List<Session> _sessions = new();
+    public List<Session> Sessions => _sessions;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<GameRunLevelChangedEvent>(MapClearing);
+        SubscribeLocalEvent<GameRuleStartedEvent>(RoundStart);
+        SubscribeLocalEvent<RoundRestartCleanupEvent>(RoundEnd);
+
+        SubscribeLocalEvent<FighterComponent, MobStateChangedEvent>(OnKillReported);
+        SubscribeLocalEvent<FighterComponent, PlayerDetachedEvent>(PlayerHasDisconnectednected);
+        SubscribeLocalEvent<FighterComponent, EraseEvent>(EraseАPlayer);
+        SubscribeLocalEvent<FighterComponent, EntityTerminatingEvent>(DeleteАPlayer);
+    }
+
+    private void RoundStart(ref GameRuleStartedEvent _)
+    {
+        CreateNewSession();
+    }
+    private void RoundEnd(RoundRestartCleanupEvent _)
+    {
+        _sessions.Clear();
+    }
+
+    public void CreateNewSession()
+    {
+        var query = EntityQueryEnumerator<CSRuleComponent, GameRuleComponent>();
+        while (query.MoveNext(out var uId, out var csRuleC, out var gRuleC))
+        {
+            if (!GameTicker.IsGameRuleActive(uId, gRuleC))
+                return;
+
+            if ((this._sessions?.Count ?? 0) < csRuleC.NumberOfSessions)
```

Comment:
```suggestion
            if ((_sessions?.Count ?? 0) < csRuleC.NumberOfSessions)
```
Не нужно. Система и так понимает

Reply:
А я думал вы любите синтаксический сахар